content/realtime/metachannels.textile

```diff
@@ -0,0 +1,110 @@
+---
+title: Metachannels
+meta_description: "Metachannels are channels that publish metadata messages, such as for channel and connection lifecycle events."
+meta_keywords: "Ably, metachannels, metadata, channel metadata, lifecycle events, connection lifecycle, channel lifecycle."
+section: realtime
+index: 5
+jump_to:
+  Help with:
+    - Overview#overview
+    - Permissions#permissions
+    - Subscribing to metachannel events#subscribing-to-metadata-events
+---
+
+h2(#overview). Overview
+
+Metachannels are a namespace of channels which all start with the @[meta]@ qualifier, differentiating them from regular channels. An example of a metachannel is @[meta]channel.lifecycle@.
+
+There are a number of metachannels available:
+
+- [meta]log := This metachannel broadcasts error messages that you would not otherwise have any way of receiving (with the exception of errors delivering push notifications, see below). For example, if Ably receives an error response when delivering a reactor webhook to your endpoint, Ably sends a message on the log metachannel with information about the error. Errors where the client can be directly notified are not sent on the log metachannel. For example, if a client attempts to publish a message but exceeds a channel rate limit, the client can be notified by an error callback "passed to publish":/realtime/channels#publish.
+- [meta]log:push := This metachannel is similar to @[meta]log@, but only for errors that happen during delivery of push notifications.
+- [meta]channel.lifecycle := This metachannel publishes messages about channel lifecycle.
+- [meta]connection.lifecycle := This metachannel publishes messages about the lifecycle of realtime connections.
+
+h2(#permissions). Permissions
+
+An Ably key has a "capability":/core-features/authentication#capabilities-explained that lists available resources and, for any given resource, a set of permitted operations. The wildcard resource '@*@' will match any regular channel name.
+
+In order to grant permission to a user to access a "metachannel":#metachannels, the resource name or names in the @capability@ must include the @[meta]@ qualifier explicitly. If you are using an API Key, you can "set capabilities in your dashboard":https://support.ably.com/solution/articles/3000073773.
```

Comment:
@SimonWoolf Interesting. I was thrown at first because the current docs say this: 
![image](https://user-images.githubusercontent.com/25511700/108068507-15cc2500-705a-11eb-80ba-3ca473b75beb.png)
However, doing some testing it looks like this is not true, and you are quite correct. If I remove "Channel Metadata" capability from the API key, but generate token requests with the wildcard capability `"capability":"{\"*\":[\"*\"]}"`, I receive metachannel events, but not inband occupancy events. I believe this is expected behaviour?

Reply:
>> Add: "You can also use a wildcard qualifier, e.g. `[*]*`, which will match all channels and all metachannels."

>  I was thrown at first because the current docs say [that `"*": ["*"]` (that is, a resource name of `*`) will not give metachannel permissions]

Both of those are correct.

- `*` as a resource name matches all normal channels, but no metachannels.
- To match both normal channels and metachannels you can use `[*]*`.
- To match all metachannels and no normal channels you can use `[meta]*`
- To match just a single metachannel you can use (say) `[meta]channel.lifecycle`

> If I ... generate token requests with the wildcard capability `"capability":"{\"*\":[\"*\"]}"`, I receive metachannel events, but not inband occupancy events. I believe this is expected behaviour?

No, `"{\"*\":[\"*\"]}"` will not give you the capability to attach to metachannels.

I'm pretty sure we don't have a bug there so I suspect you're not doing what you think you are there. PM me if you want help working out what.